MacrobenchmarkSample/macrobenchmark/src/main/kotlin/com/example/macrobenchmark/baselineprofile/ComposeActivityBaselineProfileGenerator.kt

```diff
@@ -42,16 +38,16 @@ class ComposeActivityBaselineProfileGenerator {
             maxIterations = 15,
             stableIterations = 3
         ) {
-            // Start into the Compose Activity
-            startActivityAndWait(Intent("$TARGET_PACKAGE.COMPOSE_ACTIVITY"))
+            uiAutomator {
+                // Start into the Compose Activity
+                startIntent(Intent("$TARGET_PACKAGE.COMPOSE_ACTIVITY"))
 
-            // Scrolling through the Compose journey
-            device.waitAndFind(By.res("myLazyColumn")).also {
-                it.setGestureMargin(device.displayWidth / 10)
-                it.fling(Direction.DOWN)
+                // Scrolling through the Compose journey
+                with(onView { viewIdResourceName == "myLazyColumn" }) {
+                    setGestureMargin(device.displayWidth / 10)
+                    setOf(Direction.DOWN, Direction.UP).forEach { fling(it) }
```

Comment:
Set seems weird here? Do you mean list? Set's are both unordered and if I then put another DOWN in it wouldn't work